loader/vk_loader_platform.h

```diff
@@ -24,6 +24,14 @@
  */
 #pragma once
 
+#if defined(__FreeBSD__)
+#ifndef __BSD_VISIBLE
+#define __BSD_VISIBLE 1
+#endif
+#include <sys/types.h>
+#include <sys/select.h>
+#endif
```

Comment:
This strikes me as a change that should be done in the build system. Assuming CMake is being used to build the loader on FreeBSD, if `__BSD_VISIBLE` was defined by cmake similar to how _GNU_SOURCE is defined for linux/apple platforms, then the need to put `vk_loader_platform.h` as the first include goes away. I would generally prefer that because I find that order-dependent includes cause more headaches than they are worth.

The CMake code for _GNU_SOURCE is here:
https://github.com/KhronosGroup/Vulkan-Loader/blob/61688e09fecf7f8b7749644110be9a806d4b0f4d/loader/CMakeLists.txt#L308

Reply:
indeed! thank you, I'll modify CMake!